token/services/network/fabric/lookup/deliveryllm.go

```diff
@@ -0,0 +1,185 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package lookup
+
+import (
+	"context"
+	"strings"
+
+	vault2 "github.com/hyperledger-labs/fabric-smart-client/platform/common/core/generic/vault"
+	driver2 "github.com/hyperledger-labs/fabric-smart-client/platform/common/driver"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/common/utils/collections"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/fabric"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/generic/committer"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/generic/fabricutils"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/generic/finality"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/generic/rwset"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/fabric/core/generic/vault"
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/tracing"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/network/common/rws/translator"
+	"github.com/hyperledger-labs/fabric-token-sdk/token/services/network/driver"
+	"github.com/hyperledger/fabric-protos-go/common"
+	"github.com/pkg/errors"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type newTxInfoMapper = func(network, channel string) finality.TxInfoMapper[TxInfo]
+
+type Listener interface {
+	// OnStatus is called when the key has been found
+	OnStatus(ctx context.Context, key string, value []byte)
+	// OnError is called when an error occurs during the search of the key
+	OnError(ctx context.Context, key string, err error)
+}
+
+type listenerEntry struct {
+	namespace driver2.Namespace
+	listener  Listener
+}
+
+func (e *listenerEntry) OnStatus(ctx context.Context, info TxInfo) {
+	logger.Debugf("notify info [%v] to namespace [%s]", info, e.namespace)
+	if len(e.namespace) == 0 || len(info.Namespace) == 0 || e.namespace == info.Namespace {
+		e.listener.OnStatus(ctx, info.Key, info.Value)
+	}
+}
+
+func (e *listenerEntry) Equals(other finality.ListenerEntry[TxInfo]) bool {
+	return other != nil && other.(*listenerEntry).listener == e.listener
+}
+
+type TxInfo struct {
+	Namespace driver2.Namespace
+	Key       driver2.TxID
+	Value     []byte
+}
+
+func (i TxInfo) TxID() driver2.TxID {
```

Comment:
Maybe we should rename this interface method to LookupID

Reply:
we cannot because we must `TxID` to match the expected interface